apps/desktop-wallet/src/storage/global/globalActions.ts

```diff
@@ -58,9 +60,54 @@ export const appDataCleared = createAction('app/appDataCleared')
 
 export const appDataClearFailed = createAction('app/appDataClearFailed')
 
-export const receiveTestnetTokens = createAsyncThunk<SentTransaction, AddressHash, { rejectValue: SnackbarMessage }>(
-  'assets/receiveTestnetTokens',
-  async (destinationAddress: AddressHash, { rejectWithValue, fulfillWithValue }) => {
+export const receiveFaucetTokens = createAsyncThunk<SentTransaction, AddressHash, { rejectValue: SnackbarMessage }>(
+  'assets/receiveFaucetTokens',
+  async (destinationAddress: AddressHash, { getState, rejectWithValue, fulfillWithValue }) => {
+    const state = getState() as RootState
+    const currentNetwork = state.network
+
+    if (!['testnet', 'devnet'].includes(currentNetwork.name))
+      return rejectWithValue({
+        text: i18n.t('You need to be on testnet or devnet in order to use the faucet.'),
+        type: 'alert'
+      })
+
+    const txBoilerplate: Omit<SentTransaction, 'hash'> = {
+      fromAddress: 'Faucet',
+      toAddress: destinationAddress,
+      amount: undefined,
+      timestamp: new Date().getTime(),
+      status: 'sent',
+      type: 'faucet'
+    }
+
+    if (currentNetwork.name === 'devnet') {
+      throttledClient.init(currentNetwork.settings.nodeHost, currentNetwork.settings.explorerApiHost)
+
+      const faucetPublicKey = '0381818e63bd9e35a5489b52a430accefc608fd60aa2c7c0d1b393b5239aedf6b0'
+      const faucetPrivateKey = 'a642942e67258589cd2b1822c631506632db5a12aabcf413604e785300d762a5'
```

Comment:
ahm.... @mvaivre are we should this is safe? Can't someone just drain the testnet faucet now?